Integrations/integration-Alexa.yml

```diff
@@ -0,0 +1,119 @@
+description: Alexa provides website ranking information that can be useful in determining
+  if the domain in question has a strong web presence.
+detaileddescription: To use this integration, simply call ```!awis-info```. !awis
+  accepts "url" as a parameter to query.
+configuration:
+- display: Sensitivity threshold for configuring which domains are suspicious versus
+    trusted.
+  name: threshold
+  defaultvalue: "2000000"
+  type: 0
+  required: true
+script:
+  script: |
+    import urllib, sys, re
+    import xml.etree.ElementTree as ET
+    from collections import defaultdict
+
+
+    """GLOBAL VARIABLES/CONSTANTS"""
+    THRESHOLD = int(demisto.params().get('threshold'))
+
+
+    """MAIN FUNCTIONS"""
+    def awis(domain):
+        xml = urllib.urlopen('http://data.alexa.com/data?cli=10&dat=s&url=%s'%domain).read()
+        root = ET.fromstring(xml)
+        return root
+
+
+    """COMMAND FUNCTIONS"""
+    def awis_command():
+        domain = demisto.args().get('domain')
+        root = awis(domain)
+        dbot_ec = defaultdict()
+        dom_ec = defaultdict()
+        # Wrapping in Try Except since when no results are found, endpoint returns 404
+        try:
+            domain_res = root.find("SD").attrib['HOST']
+            rank = root.find("SD[0]/POPULARITY").attrib['TEXT']
+            if int(rank) > THRESHOLD:
+                dbot_score = 2
+            else:
+                dbot_score = 0
+        except:
+            rank = 'Unknown'
+            dbot_score = 2
```

Comment:
why not 0?

Reply:
When a domain is not found, the service returns a 404 error which is why we wrap in the try except. After, we catch it, we give the scoring for it.

If a site is not found, it means that the ranking is so low, that it doesn't make the alexa list/they haven't crawled the site. We treat it as suspicious because sites fall into two categories here.

1. They are brand new - If the site is new, alexa has not collected data on it.
2. They are not generating web traffic - means the site is obscure and possibly stale, or the owner is trying to hide the site. (It's fairly easy to get a site ranked on Alexa).